java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -832,7 +835,35 @@ public final ColumnVector mergeAndSetValidity(BinaryOp mergeOp, ColumnView... co
       columnViews[i] = columns[i].getNativeView();
     }
 
-    return new ColumnVector(bitwiseMergeAndSetValidity(getNativeView(), columnViews, mergeOp.nativeId));
+    ColumnVector potentialRes = new ColumnVector(bitwiseMergeAndSetValidity(getNativeView(), columnViews, mergeOp.nativeId));
+
+    /**
+     * We need to sanitize the vector to make sure of two things
+     * 1. Nulls set in the new vector don't have non-empty nested types
+     * 2. Nulls in 'this' stay nulls otherwise bad things will happen
+     */
+    ColumnVector finalRes;
+    if (potentialRes.getType().isNestedType() || potentialRes.getType().hasOffsets() && potentialRes.hasNonEmptyNulls()) {
```

Comment:
hasNonEmptyNulls is expensive.  Would be better to use https://github.com/rapidsai/cudf/blob/6f3f50721e585b55bd263cc36926eb2c99c5f811/cpp/include/cudf/detail/copy.hpp#LL270C6-L270C29

But we don't have JNI for it.  Might be simpler to just put this part into C++ for the bitwiseMergeAndSetValidity.

Also if we return a ColumnVector with non-empty nulls in it from C++, won't an assertion error be throw? https://github.com/rapidsai/cudf/blob/6f3f50721e585b55bd263cc36926eb2c99c5f811/java/src/main/java/ai/rapids/cudf/ColumnView.java#L55

Reply:
cudf C++ API now guarantees that the output lists columns will never have non-empty nulls. The only exception is in-place set validity API.